File: C++/Programming Basics with C++/09. Exercise - While-loop/simple-tasks/ExamPreparation.cpp
#include<iostream>
#include <string>

using namespace std;

int main() {
    int failureCountLimit;
    cin >> failureCountLimit;
    cin.ignore();

    int problemsCount = 0;
    double totalGradeSum = 0;
    double failureCounter = 0;

    string taskName;
    while (true) {
        string currentTask;
        getline(cin, currentTask);
        if (currentTask == "Enough") {
            break;
        }
        taskName = currentTask;
        string gradeLikeText;
        getline(cin, gradeLikeText);
        int grade = stoi(gradeLikeText);

        if (grade <= 4) {
            failureCounter++;
        }
        if (failureCounter == failureCountLimit) {

            break;
        }
        problemsCount++;
        totalGradeSum += grade;
    }

    if (failureCounter != failureCountLimit) {
        cout.setf(ios::fixed);
        cout.precision(2);

        cout << "Average score: " << totalGradeSum / problemsCount << endl;
        cout << "Number of problems: " << problemsCount << endl;
        cout << "Last problem: " << taskName << endl;
    } else {
        cout << "You need a break, " << failureCounter << " poor grades." << endl;
    }
    return 0;
}

File: C#/Programming Basics with C#/09. While-Loop - Exercise/03. Exam Preparation/ExamPreparation.cs
using System;

class ProgExamPreparationram
{
    static void Main()
    {
        int unsatisfactoryЕstimates = int.Parse(Console.ReadLine());
        string nameOfTask = "name";
        int evaluation = 0;
        int counterUnsatisfactory = 0;
        string lastTaskName = nameOfTask;
        int sumEvalutions = 0;
        int numberProblems = 0;

        while (nameOfTask != "Enough" && unsatisfactoryЕstimates > counterUnsatisfactory)
        {
            lastTaskName = nameOfTask;
            nameOfTask = Console.ReadLine();
            if (nameOfTask != "Enough")
            {
                evaluation = int.Parse(Console.ReadLine());
                if (evaluation <= 4)
                {
                    ++counterUnsatisfactory;
                }
                sumEvalutions = sumEvalutions + evaluation;
                ++numberProblems;
            }
        }

        if (nameOfTask == "Enough")
        {
            Console.WriteLine($"Average score: {1.0 * sumEvalutions / numberProblems:F2}");
            Console.WriteLine($"Number of problems: {numberProblems}");
            Console.WriteLine($"Last problem: {lastTaskName}");
        }
        else
        {
            Console.WriteLine($"You need a break, {unsatisfactoryЕstimates} poor grades.");
        }
    }
}
